common/resources/meta_item_manager.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from common.engine.resource import Resource
+
+if TYPE_CHECKING:
+    from typing import Optional
+    from common.engine.engine import Engine
+    from common.utils.item.meta_item import MetaItem, MetaItemId
+
+
+class MetaItemManager(Resource):
+    def __init__(self, engine: Engine):
+        super().__init__(engine)
+        self.__meta_items: dict[MetaItemId, MetaItem] = {}
+
+    def register_meta_item(self, meta_item: MetaItem) -> None:
```

Comment:
Avoid class/method name repetition (`MetaItemManager.register_meta_item`)
```suggestion
    def add(self, item: MetaItem) -> None:
```